runtime/plaid-stl/src/github/mod.rs

```diff
@@ -1902,3 +1947,47 @@ pub fn remove_repo_from_team(
 
     Ok(())
 }
+
+/// Check a repo's CODEOWNERS file
+/// For more details, see https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#list-codeowners-errors
+pub fn check_codeowners_file(
+    owner: impl Display,
+    repo: impl Display,
+) -> Result<CodeownersStatus, PlaidFunctionError> {
+    extern "C" {
+        new_host_function_with_error_buffer!(github, check_codeowners_file);
+    }
+
+    const RETURN_BUFFER_SIZE: usize = 1024 * 1024; // 1 MiB
+    let mut return_buffer = vec![0; RETURN_BUFFER_SIZE];
+
+    let params = CheckCodeownersParams {
+        owner: owner.to_string(),
+        repo: repo.to_string(),
+    };
+
+    let params = serde_json::to_string(&params).unwrap();
+    let res = unsafe {
+        github_check_codeowners_file(
+            params.as_bytes().as_ptr(),
+            params.as_bytes().len(),
+            return_buffer.as_mut_ptr(),
+            RETURN_BUFFER_SIZE,
+        )
+    };
+
+    // There was an error with the Plaid system. Maybe the API is not
+    // configured.
+    if res < 0 {
+        return Err(res.into());
+    }
+
+    return_buffer.truncate(res as usize);
+
+    // This should be safe because unless the Plaid runtime is expressly trying
+    // to mess with us, this came from a String in the API module.
+    let response_body =
+        String::from_utf8(return_buffer).map_err(|_| PlaidFunctionError::InternalApiError)?;
+
+    Ok(CodeownersStatus::from(response_body))
```

Comment:
Yeah this should definitely be fallible that way mixing versions of Plaid rules and runtimes is more easily debuggable.